NetworthTypes.js

```diff
@@ -0,0 +1,7 @@
+const NetworthTypes = {
+    Normal: 'normal',
+    NonCosmetic: 'nonCosmetic',
+};
```

Comment:
Question of this as far as usage standpoint. Most likely we are going to want non-cosmetic networth in the same way we have unsoulbound networth. Is that possible without making 2 calls to `getNetworth` and `setNetworthType` in between?

Reply:
with the current implementation not. The reason for that is that for unsoulbound net worth the only thing you need to do is filter out soulbound items.
For non cosmetic net worth I did it the way that it uses the base price for items with applied skin. And this requires an item to be calculated again with just the base price. We could probably add non cosmetic in general but that would also add a bit more calculations to those who only want the normal one. But we could also just add another option to not calculate non cosmetic at the same time or an option to also add non cosmetic I dunno you tell me how you would do it lol

Reply:
I feel like a good way to go about this would be to start at the bottom and "calssify" most things. So maybe:

-  An `ItemCalculator` class that takes in item data in the constructor and has a `getNetworth` method that takes in applicable arguments. This class would mainly be used internally.
- A `PetCalculator` class, same as `ItemCalculator` but takes in pet data in the constructor.
- A `ProfileNetworthCalculator` class that has several different constructors. One that takes in the entire profile and parses inventories, one that takes in pre-parsed inventories, etc. Then it would have a `getNetworth` method that takes in applicable arguments and does the current calculation. Through the calculation, different values could be cached (in the object instance) so that if they run code like the example below, it wouldn't have to recompute tons of things:
```js
const profileNetworthCalc = ProfileNetworthCalculator(profileData);
const networth = profileNetworthCalc.getNetworth();
const nonCosmeticNetworth = profileNetworthCalc.getNonCosmeticNetworth({ onlyNetworth: true });
// or
const nonCosmeticNetworth = profileNetworthCalc.getNetworth({ nonCosmetic: true, onlyNetworth: true });
```
- Then maybe a `NetworthManager` singleton class like you have but it only tracks default values and updaters for items and prices (if asked). The different calculator classes would import the exported singleton class from this file for use of default arguments. It would define the defaults, but on the fly it would still be really nice to override these default settings (like `getNetworth({ onlyNetworth: true })`) without having to import the class instance and change the setting.
- Also the `UpdateManager` class with default values. With the singleton example below, this class would already be instantiated on module import making setup and usage really easy.

Example forced singleton from Chat:
```js
class Singleton {
  constructor() {
    if (Singleton.instance) {
      return Singleton.instance;
    }
    
    this.data = {};  // any initialization you need
    Singleton.instance = this;
  }
  
  // example method
  getData() {
    return this.data;
  }
  
  // example method
  setData(key, value) {
    this.data[key] = value;
  }
}

const singletonInstance = new Singleton();
Object.freeze(singletonInstance);

module.exports = singletonInstance;
```
Usage:
```js
const singleton = require('./path_to_singleton');

singleton.setData('key', 'value');
console.log(singleton.getData());
```

Reply:
Just some WIP ideas

Reply:
Also with the base Item/Pet/Essence/etc. calculator classes you could write an Interface/Abstract class that defines/implements a standard set of methods that all classes share to reduce code and unify usage.

Reply:
Really like that idea! @DuckySoLucky what are your opinions on that?

Reply:
### updateManager.js
```js
class UpdateManager {
    constructor() {
        if (UpdateManager.instance) {
            return UpdateManager.instance;
        }

        UpdateManager.instance = this;
    }
}
  
const updateManager = new UpdateManager();
  
module.exports = updateManager;
```

### index.js
```js
const ItemCalculator = require('./ItemCalculator');
const ProfileNetworthCalculator = require('./ProfileNetworthCalculator');
const networthManager = require('./networthManager');
const updateManager = require('./updateManager');

module.exports = {
    NetworthTypes,
    ItemCalculator, // class
    ProfileNetworthCalculator, // class
    networthManager, // class instance
    updateManager, // class instance
};
```

### test.js
```js
const { ItemCalculator, ProfileNetworthCalculator, networthManager, NetworthTypes, updateManager } = require('skyhelper-networth');

updateManager.disable();

networthManager.includeItemData = false;
networthManager.cachePrice = true;
networthManager.networthType = NetworthTypes.Normal;
networthManager.onlyNetworth = false;

const profileNetworthCalc = ProfileNetworthCalculator(profileData);
const networth = profileNetworthCalc.getNetworth();
const nonCosmeticNetworth = profileNetworthCalc.getNonCosmeticNetworth({ onlyNetworth: true });

const itemCalculator = ItemCalculator(itemData);
const itemValue = itemCalculator.getNetworth();

networthManager.updatePrices();
const superiorBabySkinPrice = networthManager.prices["SUPERIOR_BABY"];
```